db/model/roomType.js

```diff
@@ -123,34 +123,41 @@ module.exports = function roomType(seq, dataTypes) {
             resolve(inst);
           }
 
-          inst._previousDataValues.bots.forEach((botName) => {
-            seq.models.Bot.findOne({
-              where: {
-                name: {
-                  $iLike: botName,
-                },
-              },
-            })
-            .then((o) => {
-              inst.removeBots(o)
-              .catch(reject);
-            });
-          });
-
-          inst.dataValues.bots.forEach((botName) => {
-            seq.models.Bot.findOne({
-              where: {
-                name: {
-                  $iLike: botName,
+          new Promise((resolve, reject) => {
+            if (inst._previousDataValues.bots) {
+              inst._previousDataValues.bots.forEach((botName) => {
+                seq.models.Bot.findOne({
+                  where: {
+                    name: {
+                      $iLike: botName,
+                    },
+                  },
+                })
+                .then((o) => {
+                  inst.removeBots(o)
+                  .catch(reject);
+                });
+              });
+            }
+
+            resolve();
```

Comment:
This resolve means we don't proceed to the "then" on line 145... is that the intention? In that case, the promise could resolve empty or if goes all the way down to line 159 it resolves with the instance.